### PR TITLE
Add Mt-Bench  datasets + add operators 

### PR DESCRIPTION
**Pull request by OfirArviv**

### CODE REVIEW

File: src/unitxt/operators.py [hunk 1]
```diff
@@ -1293,6 +1313,52 @@ def _is_required(self, instance: dict) -> bool:
         return True
 
 
+class FilterByConditionBasedOnFields(FilterByCondition):
+    """Filters a stream, yielding only instances in which the values in required fields follow the required condition operator. The value is another field in the dataset.
```
Comment (yoavkatz) [inline src/unitxt/operators.py:1317]: The description not clear (and seems to by mostly copied from the FilterByConditition).

File: src/unitxt/operators.py [hunk 2]
```diff
@@ -1743,6 +1809,61 @@ def process(self, stream: Stream, stream_name: Optional[str] = None) -> Generato
         yield from stream
 
 
+class JoinStreams(MultiStreamOperator):
+    """Merges multiple streams into a single stream.
```
Comment (yoavkatz) [inline src/unitxt/operators.py:1813]: Need to update the doc - and most important add instructions.

File: src/unitxt/operators.py [hunk 3]
```diff
@@ -2104,3 +2225,13 @@ def verify(self):
                 f"If given, duplication_index_field must be a string. "
                 f"Got: {self.duplication_index_field}"
             )
+
+
+class DeleteSplits(MultiStreamOperator):
```
Comment (yoavkatz) [inline src/unitxt/operators.py:2230]: Thank for adding these operators. They will be useful in many case.
I think we need to add tests for the new operators.

File: src/unitxt/operators.py [hunk 4]
```diff
@@ -1743,6 +1809,61 @@ def process(self, stream: Stream, stream_name: Optional[str] = None) -> Generato
...
+        left_on = self.left_on if self.left_on is not None else []
+        right_on = self.right_on if self.right_on is not None else []
+        on_cols = set(on + left_on + right_on)
+        col_to_remove = list(common_cols - on_cols)
+        left_stream_df = left_stream_df.drop(columns=col_to_remove, errors="ignore")
+        right_stream_df = right_stream_df.drop(columns=col_to_remove, errors="ignore")
+
+        # TODO: We remove 'recipe_metadata'. Is that ok?
+
+        merged_df = pd.merge(
```
[the remark this answers is not in the document]
Comment (OfirArviv) [reply src/unitxt/operators.py:1851]: panda is a requirement for hugginface datasets, is that not enough?
https://github.com/huggingface/datasets/blob/b442aa2d3efc83ba0dc369adaa63cc496e3d9836/setup.py#L123

**Review by elronbandel — CHANGES_REQUESTED** [review]
Following @yoavkatz we need to add:

1) clear documentations for new operators

2) tests for new operators 

3) put stream related operators in a seperate file streams_operators.py

File: src/unitxt/operators.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (OfirArviv) [reply src/unitxt/operators.py:1857]: @elronbandel confirm its ok

File: src/unitxt/operators.py (hunk 1 above)
Comment (OfirArviv) [reply src/unitxt/operators.py]: improved it a bit

File: src/unitxt/operators.py (hunk 2 above)
Comment (OfirArviv) [reply src/unitxt/operators.py]: wdym instructions? I followed the example of the merge operator

File: prepare/cards/mt_bench/common.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    ),
+    RenameFields(
+        field_to_field={"choices": "model_output"}, apply_to_streams=["model_answer"]
+    ),
+    Apply(
+        "model_id",
+        function=str.lower,
+        to_field="model_id",
+        apply_to_streams=["judgment", "model_answer"],
+    ),
```
Comment (elronbandel) [inline prepare/cards/mt_bench/common.py:34]:
```suggestion
    Apply(
        "model_id",
        function="str.lower",
        to_field="model_id",
        apply_to_streams=["judgment", "model_answer"],
    ),
```

File: prepare/cards/mt_bench/common.py [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            "model_id": {
+                "vicuna-13b-hao-0515": "vicuna-13b-v1.3",
+                "vicuna-30b-gpt4": "vicuna-33b-v1.3",
+            }
+        },
+        strict=False,
+        apply_to_streams=["judgment", "model_answer"],
+    ),
+    CopyFields(
+        field_to_field={"model_output/0/turns": "model_output"},
```
Comment (elronbandel) [inline prepare/cards/mt_bench/common.py:46]:
```suggestion
        field="model_output/0/turns"
        to_field= "model_output",
```

**Review by elronbandel — APPROVED** [review]

**Pull request merged**